### PR TITLE
Fix bug for apply penalty on remove liquidity

### PR DESCRIPTION
**Pull request by grandizzy**
- Fix apply penalty on remove quote token even if no collateral pledged to pool
- Add test to expose scenario where lender cannot withdraw all their quote tokens if there is still collateral pledged in bucket after repay

### CODE REVIEW

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/libraries/PoolUtils.sol [hunk 1]
```diff
@@ -103,13 +103,16 @@ library PoolUtils {
...
-            uint256 ptp = Maths.wdiv(poolState_.accruedDebt, poolState_.collateral);
-            bool applyPenalty = indexToPrice(fromIndex_) > ptp;
-            if (toIndex_ != 0) applyPenalty = applyPenalty && indexToPrice(toIndex_) < ptp;
-            if (applyPenalty) {
-                amountWithPenalty_ =  Maths.wmul(amountWithPenalty_, Maths.WAD - feeRate(poolState_.rate));
+        if (depositTime_ != 0 && block.timestamp - depositTime_ < 1 days) {
+            bool applyPenalty = toIndex_ == 0; // remove quote token scenario, apply penalty
+            if (!applyPenalty) {
+                // move quote token between buckets scenario, check if moved below PTP
+                if (poolState_.collateral != 0) {
```
Comment (EdNoepel) [inline src/libraries/PoolUtils.sol]: This logic still doesn't look right.  If it's called from `removeQuoteToken` (`toIndex_ == 0`), it should not apply the penalty if withdrawal is below the PTP.

**Review by EdNoepel — CHANGES_REQUESTED** [review]

File: src/libraries/PoolUtils.sol (hunk 1 above)
Comment (grandizzy) [reply src/libraries/PoolUtils.sol]: I see. So what's the PTP value when there's no collateral pledged in pool, is it considered 0?

Comment (EdNoepel) [reply src/libraries/PoolUtils.sol]: Yes; from the spec:
```
Pool Threshold Price (PTP)
   Total Debt / Total Collateral Posted
   If no collateral has been deposited, the PTP is defined as 0.
```

Comment (grandizzy) [reply src/libraries/PoolUtils.sol]: 👍

Comment (grandizzy) [reply src/libraries/PoolUtils.sol]: pls check with https://github.com/ajna-finance/contracts/pull/334/commits/052a104966464374018811ebccdf90ca34abaee3

**Review by EdNoepel — APPROVED** [review]
Perfecto.

**Pull request merged**